src/installer.ts

```diff
@@ -2,187 +2,156 @@
+      throw new Error(
+        `'dotnet-version' was supplied in invalid format: ${this.inputVersion}! Supported syntax: A.B.C, A.B, A.B.x, A, A.x`
+      );
+    }
+    if (semver.valid(this.inputVersion)) {
+      this.resolvedArgument.type = 'version';
+      this.resolvedArgument.value = this.inputVersion;
+    } else {
+      const [major, minor] = this.inputVersion.split('.');
+
+      if (this.isNumericTag(major)) {
+        this.resolvedArgument.type = 'channel';
+        if (this.isNumericTag(minor)) {
+          this.resolvedArgument.value = `${major}.${minor}`;
+        } else {
+          const httpClient = new hc.HttpClient('actions/setup-dotnet', [], {
+            allowRetries: true,
+            maxRetries: 3
+          });
+          this.resolvedArgument.value = await this.getReleasesJsonUrl(
+            httpClient,
+            [major, minor]
+          );
+        }
+      }
+      this.resolvedArgument.qualityFlag = +major >= 6 ? true : false;
+    }
   }
-}
 
-export class DotnetCoreInstaller {
-  constructor(version: string, includePrerelease: boolean = false) {
-    this.version = version;
-    this.includePrerelease = includePrerelease;
+  private isNumericTag(versionTag): Boolean {
+    return /^\d+$/.test(versionTag);
   }
 
-  public async installDotnet() {
-    let output = '';
-    let resultCode = 0;
-
-    let calculatedVersion = await this.resolveVersion(
-      new DotNetVersionInfo(this.version)
-    );
-
-    var envVariables: {[key: string]: string} = {};
-    for (let key in process.env) {
-      if (process.env[key]) {
-        let value: any = process.env[key];
-        envVariables[key] = value;
-      }
+  public async createDotNetVersion(): Promise<{
+    type: string;
+    value: string;
+    qualityFlag: boolean;
+  }> {
+    await this.resolveVersionInput();
+    if (!this.resolvedArgument.type) {
+      return this.resolvedArgument;
     }
     if (IS_WINDOWS) {
-      let escapedScript = path
-        .join(__dirname, '..', 'externals', 'install-dotnet.ps1')
-        .replace(/'/g, "''");
-      let command = `& '${escapedScript}'`;
-      if (calculatedVersion) {
-        command += ` -Version ${calculatedVersion}`;
-      }
-      if (process.env['https_proxy'] != null) {
-        command += ` -ProxyAddress ${process.env['https_proxy']}`;
-      }
-      // This is not currently an option
-      if (process.env['no_proxy'] != null) {
-        command += ` -ProxyBypassList ${process.env['no_proxy']}`;
-      }
+      this.resolvedArgument.type =
+        this.resolvedArgument.type === 'channel' ? '-Channel' : '-Version';
+    } else {
+      this.resolvedArgument.type =
+        this.resolvedArgument.type === 'channel' ? '--channel' : '--version';
+    }
+    return this.resolvedArgument;
+  }
 
-      // process.env must be explicitly passed in for DOTNET_INSTALL_DIR to be used
-      const powershellPath = await io.which('powershell', true);
+  private async getReleasesJsonUrl(

```

Comment:
It looks like this function returns a `major.minor` version string, not a URL?
```suggestion
  private async getLatestVersion(
```